articles/tutorials/advanced/2d_shaders/07_sprite_vertex_effect/index.md

```diff
@@ -0,0 +1,863 @@
+    effect: _3dMaterial.Effect);
+```
+
+When the game runs, the text will be missing. This is because we never created a projection matrix to assign to the `MatrixTransform` shader parameter. Add this code when loading the material:
+
+```csharp
+// Load the 3d effect 
+_3dMaterial = Content.WatchMaterial("effects/3dEffect");
+_3dMaterial.IsDebugVisible = true;
+
+// this matrix code is taken from the default vertex shader code.
+Matrix.CreateOrthographicOffCenter(
+	left: 0, 
+	right: Core.GraphicsDevice.Viewport.Width, 
+	bottom: Core.GraphicsDevice.Viewport.Height, 
+	top: 0, 
+	zNearPlane: 0, 
+	zFarPlane: -1, 
+	out var projection);
+_3dMaterial.SetParameter("MatrixTransform", projection);
+```
+
+And now you should see the text normally again.
+
+![Figure 7.1: The main menu, but rendered with a custom vertex shader](./images/basic.png)
+
+### Making it Move
+
+As a quick experiment, we can show that the vertex shader can indeed modify the vertex positions further if we want to. For now, add a temporary shader parameter called `DebugOffset`:
+
+```hlsl
+float2 DebugOffset;
+```
+
+And change the vertex shader to add the `DebugOffset` to the `output.Position` after the clip-space conversion:
+
+```hlsl
+output.Position = mul(input.Position, MatrixTransform);  
+output.Position.xy += DebugOffset;
+```
+
+The sprites now move around as we adjust the shader parameter values. 
+![Figure 7.2: We can control the vertex positions](./gifs/basic.gif)
+
+It is important to build intuition for the different coordinate systems involved. Instead of adding the `DebugOffset` _after_ the clip-space conversion, if you try to add it _before_, like in the code below:
+
+```hlsl
+float4 pos = input.Position;  
+pos.xy += DebugOffset;  
+output.Position = mul(position, MatrixTransform);
+```
+
+Then you will not see much movement at all. This is because the `DebugOffset` values only go from `0` to `1`, and in world space, this really only amounts to a single pixel. In fact, exactly how much an addition of _`1`_ happens to make is entirely defined _by_ the conversion to clip-space. The `projection` matrix we created treats world space coordinates with an origin around the screen's center, where 1 unit maps to 1 pixel. Sometimes this is exactly what you want, and sometimes it can be confusing. 
+
+![Figure 7.3: Changing coordinates before clip-space conversion](./gifs/basic-2.gif)
+
+### Perspective Projection
+
+The world-space vertices can have their `x` and `y` values modified in the vertex shader, but what about the `z` component? The orthographic projection essentially _ignores_ the `z` component of a vertex and treats all vertices as though they are an equal distance away from the camera. If you change the `z` value, you may _expect_ the sprite to appear closer or further away from the camera, but the orthographic projection matrix does not do that. 
+
+To check, try modify the shader code to adjust the `z` value based on one of the debug values:
+
```

Comment:
Link to doc that talks about the Orthographic clipping clane, for example:
https://docs.monogame.net/articles/getting_to_know/whatis/graphics/WhatIs_Camera.html